powershell/plugins/plugin-tweak-model.ts

```diff
@@ -75,86 +75,97 @@ async function tweakModelV2(state: State): Promise<PwshModel> {
+    parameters: new Dictionary<any>(),
+  };
+
+  // we're going to create a schema that represents the distinct sum 
+  // of all operation PATH parameters
   const universalId = new ObjectSchema(`${serviceName}Identity`, 'Resource Identity');
   universalId.apiVersions = universalId.apiVersions || [];
   state.model.schemas.objects = state.model.schemas.objects || [];
-  (<any>universalId.language.default).uid = true;
+  state.model.schemas.objects.push(universalId);
 
-  universalId.properties = universalId.properties || new Array<Property>();
-  const urlParam = new Set<string>();
-  for (const operationGroup of values(model.operationGroups)) {
-    for (const operation of values(operationGroup.operations)) {
-      for (const param of values(operation.parameters).where(each => each.protocol.http?.in === ParameterLocation.Path)) {
+  for (const group of values(model.operationGroups)) {
+    for (const operation of values(group.operations)) {
+      for (const param of values(operation.parameters).where(each => each.protocol?.http?.in === ParameterLocation.Path)) {
         const name = param.language.default.name;
+        const hasName = universalId.properties?.findIndex((prop) => prop.language.default.name === name);
+        if (!hasName) {
+          if (!universalId.properties) {
+            universalId.properties = [];
+          }
+          const newProp = new Property(name, param.language.default.description, param.schema);
+          newProp.required = false;
+          newProp.readOnly = false;
+          newProp.serializedName = name;
+          universalId.properties.push(newProp);
+        }
+      }
+    }
+  }
 
-        if (!urlParam.has(name)) {
-          urlParam.add(name);
-          universalId.addProperty(new Property(name, '', param.schema, {
-            serializedName: name, language: {
-              default: {
-                description: param.summary,
-                name: name,
-                required: false,
-                readOnly: false,
-                uid: 'universal-parameter:resource identity'
-              }
-            }
-          }));
+  // identify models that are polymorphic in nature
+  for (const schema of allSchemas) {
+    if (schema instanceof ObjectSchema) {
+      const objSchema = schema as ObjectSchema;
+      // if this actual type is polymorphic, make sure we know that.
+      // parent class
+      if (objSchema.discriminator) {
+        objSchema.language.default.isPolymorphic = true;
+        if (objSchema.children) {
+          objSchema.language.default.polymorphicChildren = objSchema.children?.all;
         }
       }
+
+      // sub class
+      if (objSchema.discriminatorValue) {
+        objSchema.language.default.discriminatorValue = objSchema.extensions?.['x-ms-discriminator-value'];
+      }
     }
   }
 
-  if (await state.getValue('azure', false)) {
-    universalId.addProperty(new Property('id', '', new StringSchema('_identity_type_', 'Resource identity path'),
```

Comment:
We need to keep this part to support viaIdentity for azure

Reply:
Got it. Add it back